### ISSUE TITLE
[HEADLESS] Impossible to start headless script on Ubuntu 18.04 LTS

### ISSUE THREAD

**Issue by RoiArthurB** [gama-platform/gama#2773]
**Describe the bug**
Impossible to start `./headless/gama-headless.sh` on Ubuntu 18.04 LTS. Here's the output when I run the script

```
$ sh headless/gama-headless.sh -h
headless/gama-headless.sh: 5: headless/gama-headless.sh: declare: not found
headless/gama-headless.sh: 6: headless/gama-headless.sh: declare: not found
headless/gama-headless.sh: 14: headless/gama-headless.sh: Bad substitution
```

The bug is about how some variables are set with an unknown keyword `declare`. And so the script return an error because variables are not set.

**Expected behavior**

```
$ sh headless/gama-headless.sh -h
******************************************************************
* GAMA version 1.8                                               *
* http://gama-platform.org                                       *
* (c) 2007-2019 UMI 209 UMMISCO IRD/SU & Partners                *
******************************************************************



sh ./gama-headless.sh [Options] [XML Input] [output directory]
	
List of available options:
 -help     				-- get the help of the command line
 -version     				-- get the the version of gama
 -m [mem]    				-- allocate memory, eg. 2048m
 -c        				-- start the console to write xml parameter file
 -v 					-- verbose mode
 -hpc [core] 				-- set the number of core available for experimentation
 -socket [socketPort] 			-- start socket pipeline to interact with another framework
 -p                                     -- start pipeline to interact with another framework
 -validate [directory]                  -- invokes GAMA to validate the models present in the directory passed as argument
 -test [directory]		   	-- invokes GAMA to execute the tests present in the directory and display their results
 -failed		      	        -- only display the failed and aborted test results
 -xml	[experimentName] [modelFile.gaml] [xmlOutputFile.xml]
				        --  build an xml parameter file from a model

```

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04 LTS (up to date)
 - GAMA version: 1.8 Continuous Build w/ JDK && w/o JDK
 - Java version: OpenJDK 8 && Oracle JDK 8

**Workaround**
I've tried the headless with Manjaro 18 and the script is working.
So I've tried to find differencies and the bash version are differents : 

*Ubuntu*
```
$ bash --version
GNU bash, version 4.4.19(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

*Manjaro*
```
$ bash --version
GNU bash, version 5.0.7(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2019 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

The bug is caused by the `declare -i` keyword. I didn't find any information about it, but maybe it had been introduced in v5 ?

**Comment by RoiArthurB** [comment]
Okay I've find what the problem is : 
The script is in ***bash*** and not in ***shell***  ( [here](https://stackoverflow.com/questions/5725296/difference-between-sh-and-bash)'s more information about the difference).

So many updates should be applied : 
- First, the script should be run this the command `bash` and not with `sh` command. So running the script should be : 
```
$ bash headless/gama-headless.sh 
bash headless/gama-headless.sh 
headless/gama-headless.sh
headless/gama-headless.sh: line 81: -1: bad substitution
******************************************************************
* GAMA version 1.8                                               *
* http://gama-platform.org                                       *
* (c) 2007-2019 UMI 209 UMMISCO IRD/SU & Partners                *
******************************************************************
The input or output file are not specied. Please check the path of your files and output file.
Use the help for more information: ./gama-headless -help
```
- Second, the shebang should be updated to indicate bash (`#!/bin/bash`) and not shell (`#!/bin/sh`). In fact, if we execute the script, the script will be launched we the command indicated in the shebang.
```
$ chmod +x headless/gama-headless.sh
$ ./headless/gama-headless.sh 
./headless/gama-headless.sh 
./headless/gama-headless.sh: line 81: -1: bad substitution
******************************************************************
* GAMA version 1.8                                               *
* http://gama-platform.org                                       *
* (c) 2007-2019 UMI 209 UMMISCO IRD/SU & Partners                *
******************************************************************
The input or output file are not specied. Please check the path of your files and output file.
Use the help for more information: ./gama-headless -help
```

PS: the error message is just because I didn't set any argument in my command (like `-help`)

**Comment by RoiArthurB** [comment]
Fixed  7658e47

**Comment by chintanp** [comment]
For anyone facing issues with running the examples on Ubuntu etc., the headless README needs to be updated. What worked for me is this: 

`./gama-headless.sh <path_to_xml> <path_tro_output>`
